src/Command/Whowas.js

```diff
@@ -0,0 +1,100 @@
+const Command = require('../Command');
+
+const uuidRegEx = RegExp('^[0-9a-f]{8}-?[0-9a-f]{4}-?[1-5][0-9a-f]{3}-?[89ab][0-9a-f]{3}-?[0-9a-f]{12}$', 'i');
+const mcidRegEx = RegExp('^[abcdefghijklmnopqrstuvwxyz0123456789_]{3,16}$', 'i');
+const messages = {
+    error: "Uh oh, there seems to be an error retrieving data from the Mojang API with that input. Verify your input or try again later!",
+    invalid: "It seems like you haven't provided a valid username or UUID. Please verify your input.",
+    names: "I have retrieved the following information about `{input}`:\n{names}",
+    originalName: "(original)",
+    changedName: "(changed to at {date})"
+};
+
+module.exports = Command.extend({
+    commandName: 'whowas',
+    https: null,
+    moment: null,
+    dependencies: {
+        'https': 'https',
+        'moment': 'moment'
+    },
+    processMessage: function (message, tokens) {
+        let i18n = this.i18n;
+        let identifier = tokens[1];
+
+        if (mcidRegEx.test(identifier)) {
+            this.fetchData(`https://api.mojang.com/users/profiles/minecraft/${identifier}`, (res) => {
+                if (res === false) {
+                    message.channel.send(i18n.__mf(messages.error));
+                    return;
+                }
+
+                identifier = res.id.replace('-', ''); // Remove any dashes from the UUID
+
+                this.fetchData(`https://api.mojang.com/user/profiles/${identifier}/names`, (res) => {
+                    if (res === false) {
+                        message.channel.send(i18n.__mf(messages.error));
+                        return;
+                    }
+
+                    message.channel.send(i18n.__mf(messages.names, { input: tokens[1], names: this.processNames(res) }));
+                });
+            });
+        } else if (uuidRegEx.test(identifier)) {
+            identifier = identifier.replace('-', ''); // Remove any dashes from the UUID
+
+            this.fetchData(`https://api.mojang.com/user/profiles/${identifier}/names`, (res) => {
+                if (res === false) {
+                    message.channel.send(i18n.__mf(messages.error));
+                    return;
+                }
+
+                message.channel.send(i18n.__mf(messages.names, { input: tokens[1], names: this.processNames(res) }));
+            });
+        } else {
+            message.channel.send(i18n.__mf(messages.invalid));
+        }
+    },
+    processNames: function(names) {
+        let nameHistory = [];
+        names.map(entry => {
+            if (entry.changedToAt) {
+                nameHistory.push(`- ${entry.name} ${this.i18n.__mf(messages.changedName, { date: this.moment(entry.changedToAt).format('ll') })}\n`);
+            } else {
+                nameHistory.push(`- ${entry.name} ${this.i18n.__mf(messages.originalName)}\n`);
+            }
+        });
+
+        return nameHistory.join('');
```

Comment:
Since each line in nameHistory is a literal line, you could leave off the \n for each "push" and join with \n

But that's not a necessary change - just something I wanted to note